pkg/exporter/process_test.go

```diff
@@ -247,13 +247,21 @@ func TestExportingProcess_SendingDataRecordToLocalTCPServer(t *testing.T) {
 	dataRecBytes := dataRecBuff.Bytes()
 
 	bytesSent, err := exporter.SendSet(dataSet)
-	if err != nil {
-		t.Fatalf("Got error when sending record: %v", err)
-	}
+	assert.NoError(t, err)
 	// 28 is the size of the IPFIX message including all headers (20 bytes)
 	assert.Equal(t, 28, bytesSent)
 	assert.Equal(t, dataRecBytes, <-buffCh)
 	assert.Equal(t, uint32(1), exporter.seqNumber)
+
+	// Create data set with multiple data records to test invalid message length
+	// logic for TCP transport.
+	dataSet = entities.NewSet(entities.Data, templateID, false)
+	for i := 0; i < 10000; i++ {
+		dataSet.AddRecord(elements, templateID)
+	}
+	bytesSent, err = exporter.SendSet(dataSet)
+	assert.Error(t, err)
```

Comment:
I tried to log the error here. the error message is `could not send the complete message on the connection` instead of exceeding maximum size. Not sure whether we can test it here since 65535 is rather large and current msgLen for 10000 records is only 14484

Reply:
Good catch with the error. Each data record is around ~8 bytes, so the number of data records added is fine but uint16 casting in `set.GetBufferLen` is causing the issue. Fixed it.